spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/web/SpringDataWebAutoConfiguration.java

```diff
@@ -79,4 +82,10 @@ public SortHandlerMethodArgumentResolverCustomizer sortCustomizer() {
 		return (resolver) -> resolver.setSortParameter(this.properties.getSort().getSortParameter());
 	}
 
+	@Primary // override bean created by @EnableSpringDataWebSupport
```

Comment:
I think we should provide a better way to do this, perhaps requiring a change in Spring Data, as I'd prefer not to have multiple beans of the same type in the context.

Reply:
I agree.

Reply:
@odrotbohm I have a vague recollection that we've already discussed this but I haven't been able to find it.

I guess would could use `@ConditionalOnProperty` to either auto-configure `@EnableSpringDataWebSupport` or `@EnableSpringDataWebSupport(pageSerializationMode = VIA_DTO)`. I'm a little hesitant as this approach won't scale as more attributes are added but it would work for now I think.

Reply:
We ultimately want to nudge people into using `VIA_DTO` at some point, as that's the better way of serializing `Page` instances by default. We kept the default at `DIRECT` for 2024.0.0 to not force applications into the new model, potentially causing them to break their APIs and thus not upgrading. That said, if the Boot team was fine with such a change, we could of course already flip the default, with the users being able to re-instantiate the old behavior by explicitly adding `@EnableSpringDataWebSupport(pageSerializationMode = DIRECT)` to their application.

Reply:
`@ConditionalOnProperty` is not appropriate here, It's just a flag that should be stick to other pageable settings as possible, not like `spring.data.redis.client-type` which affect creation of other beans.
I'd prefer to not register `SpringDataWebSettings` bean if `pageSerializationMode` is default as `SpringDataWebAutoConfiguration` did, then the `@Primary` could be removed.